src/mixins/filing-template-mixin.ts

```diff
@@ -611,4 +561,78 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
     }
     this.setBusinessContact(contactPoint)
   }
+
+  /** Build Staff Payment data into the filing.
+   * @param filing The alteration or correction filing.
+   */
+  private buildStaffPayment (filing: AlterationFilingIF | CorrectionFilingIF): any {
+    // Populate Staff Payment according to payment option
+    switch (this.getStaffPayment.option) {
+      case StaffPaymentOptions.FAS:
+        filing.header.routingSlipNumber = this.getStaffPayment.routingSlipNumber
+        filing.header.priority = this.getStaffPayment.isPriority
+        break
+
+      case StaffPaymentOptions.BCOL:
+        filing.header.bcolAccountNumber = this.getStaffPayment.bcolAccountNumber
+        filing.header.datNumber = this.getStaffPayment.datNumber
+        filing.header.folioNumber = this.getStaffPayment.folioNumber // this overrides original folio number
+        filing.header.priority = this.getStaffPayment.isPriority
+        break
+
+      case StaffPaymentOptions.NO_FEE:
+        filing.header.waiveFees = true
+        filing.header.priority = false
+        break
+
+      case StaffPaymentOptions.NONE: // should never happen
+        break
+    }
+
+    return filing
```

Comment:
Theoretically this isn't needed, because the code above changes the original object. (If you go this route, you'd have to update lines 170 and 267.)

Reply:
The object is passed by reference I think (like a pointer to an object in C).

Reply:
Or maybe not.
https://stackoverflow.com/questions/13104494/does-javascript-pass-by-reference

Reply:
https://www.geeksforgeeks.org/pass-by-value-and-pass-by-reference-in-javascript/

I'm going to assume the above are correct and resolve this conversation with no change required ;)

Reply:
Hmm.. i thought it would be. 
If it isn't it would be cleaner and preferable. I will try!

Reply:
Yea.. works good! I was worried the filing would return without the changes.
Good clean up, thanks!

Reply:
OK so now I'm confused why it works 🤷

Reply:
Like you said, the javascript reference :) 

Tbh i dunno either, it does mutate the original values though.

Reply:
Maybe this?
https://dev.to/bytebodger/a-gotcha-of-javascript-s-pass-by-reference-1afm

Reply:
TLDR